City-of-Helsinki/kukkuu-admin
# KK-630 | Add event groups resource

## Description

This PR adds the event groups resources, changes the event list so that it's provided by the events and event groups resource and refactors authentication.

Initially I envisioned this view structure:

![image](https://user-images.githubusercontent.com/9090689/98916047-5ce2d100-24d3-11eb-8f13-4cc9d7926f3a.png)

This means that event groups will make use of some event views in order to provide users the UI to complete event specific user goals.

I've also added a "pseudo resource" called "event and event groups". This may be too defensive, but react-admin assumes that the list API  yields resources of a specific type and may use it as a fallback with some input types for instance. On the other hand the `List` component is tightly integrated into this endpoint. By using a custom resource, we can guarantee that no event select (etc) will stop working because the API now returns two kinds of entities in the same list and we are able to use react-admin's `List` component without much custom logic.

I ported the auth changes from youth-membership-admin-ui into this application as well.

## Context

This change allows the other changes needed for event groups to be developed in a more concurrent fashion.

[KK-630](https://helsinkisolutionoffice.atlassian.net/browse/KK-630)

## How Has This Been Tested?

I've added tests for the authentication changes. The resource will be e2e tested once it's more complete.

## Manual Testing Instructions for Reviewers

As an admin user

1. Login
1. Go to event list
1. Click on event
1. Expect to be taken to the event in quesion
